app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java

```diff
@@ -841,5 +863,40 @@ private PlayQueue getPlayQueue(final int index) {
         }
         return new SinglePlayQueue(streamInfoItems, index);
     }
+
+    private void createShareConfirmationDialog() {
+        if (getContext() == null) {
+            return;
+        }
+
+        final DialogSharePlaylistBinding dialogBinding = DialogSharePlaylistBinding
+                .inflate(getLayoutInflater());
+
+        new AlertDialog.Builder(getContext())
+                .setTitle(R.string.share_playlist)
+                .setCancelable(true)
+                .setView(dialogBinding.getRoot())
+                .setPositiveButton(R.string.share_playlist_with_details, (dialog, which) -> {
+                    sharedPreferences.edit()
+                            .putBoolean(requireContext().getString(
+                                    R.string.share_playlist_with_details_can_show_dialog_key),
+                                    !dialogBinding.rememberChoiceCheckBox.isChecked())
+                            .putBoolean(requireContext().getString(
+                                    R.string.share_playlist_with_details_key), true)
+                            .commit();
```

Comment:
```suggestion
                            .apply();
```

Reply:
I had used `commit` as we need to make the changes synchronously for the `sharePlaylist` method. I have  tested with `apply` and it seems to work properly.